src/Npgsql.EntityFrameworkCore.PostgreSQL/Migrations/Internal/NpgsqlHistoryRepository.cs

```diff
@@ -85,22 +85,33 @@ protected override string ExistsSql
 
         public override string GetCreateIfNotExistsScript()
         {
-            return GetCreateScript();
+            var script = GetCreateScript();
+            return script.Insert(script.IndexOf("CREATE TABLE", StringComparison.Ordinal) + 12, " IF NOT EXISTS");
         }
 
         public override string GetBeginIfNotExistsScript(string migrationId)
```

Comment:
Sorry to be jerking you around like this.

If you put the tableId expression directly inside the string (without using the tableId variable), you can also convert the method into an expression-bodied method (no curly braces, just =>). After that I think it'll be ready for merging.